pulsar-common/src/main/java/com/yahoo/pulsar/common/util/collections/ConcurrentLongPairSet.java

```diff
@@ -528,4 +529,23 @@ public int compareTo(LongPair o) {
             }
         }
     }
+    
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append('{');
+        final AtomicBoolean first = new AtomicBoolean(true);
```

Comment:
Though, what you could do here is to always put the ", " after the current iterator, and always remove it at the end, by trimming the size: 

```java
...
sb.setLength(sb.length() - 2);
```